vpr/src/base/vpr_constraints.cpp

```diff
@@ -5,6 +5,10 @@ void VprConstraints::add_constrained_atom(const AtomBlockId blk_id, const Partit
 
     auto got = constrained_atoms.find(blk_id);
 
+    /**
+     * Each atom can only be in one partition. If the atoms already has a partition id assigned to it,
```

Comment:
If the *atom* already has a partition id assigned to it the *id* will be swithched ...